tla/consensus/ccfraft.tla

```diff
@@ -459,45 +459,28 @@ PlausibleSucessorNodes(i) ==
-    /\ currentTerm = [i \in servers |-> IF i = startNode THEN StartTerm ELSE 0]
-    /\ state       = [i \in servers |-> IF i = startNode THEN Leader ELSE None]
-    /\ votedFor    = [i \in servers |-> Nil]
-    /\ log         = [i \in servers |-> IF i = startNode
-                                        THEN << [term |-> StartTerm, contentType |-> TypeReconfiguration, configuration |-> {startNode}],
-                                                [term |-> StartTerm, contentType |-> TypeSignature] >>
-                                        ELSE << >>]
-    /\ commitIndex  = [i \in servers |-> IF i = startNode THEN Len(log[i]) ELSE 0]
-    /\ committableIndices  = [i \in servers |-> {}]
-
-\* Generate initial state for a startNodes configuration
-JoinedState(startNodes, servers) ==
+StartLog(startNode, nextNodes) ==
+    << [term |-> StartTerm, contentType |-> TypeReconfiguration, configuration |-> startNode],
+       [term |-> StartTerm, contentType |-> TypeSignature] >>
+
+InitLogConfigServerVars(startNodes, xlog(_,_)) ==
     /\ reconfigurationCount = 0
     /\ removedFromConfiguration = {}
-    /\ \E startNode \in startNodes:
-        /\ configurations = [ i \in servers |-> IF i \in startNodes  THEN (3 :> startNodes) ELSE << >>]
-        /\ currentTerm = [i \in servers |-> IF i \in startNodes THEN StartTerm ELSE 0]
-        /\ state       = [i \in servers |-> IF i = startNode THEN Leader ELSE IF i \in startNodes THEN Follower ELSE None]
-        /\ votedFor    = [i \in servers |-> Nil]
-        /\ log         = [i \in servers |-> IF i \in startNodes
-                                            THEN << [term |-> StartTerm, contentType |-> TypeReconfiguration, configuration |-> {startNode}],
-                                                    [term |-> StartTerm, contentType |-> TypeSignature],
-                                                    [term |-> StartTerm, contentType |-> TypeReconfiguration, configuration |-> startNodes],
-                                                    [term |-> StartTerm, contentType |-> TypeSignature] >>
-                                            ELSE << >>]
-        /\ commitIndex  = [i \in servers |-> IF i \in startNodes THEN Len(log[i]) ELSE 0]
-        /\ committableIndices  = [i \in servers |-> {}]
-
+    /\ committableIndices  = [i \in Servers |-> {}]
+    /\ votedFor    = [i \in Servers |-> Nil]
+    /\ currentTerm = [i \in Servers |-> IF i \in startNodes THEN StartTerm ELSE 0]
+    /\ \E sn \in startNodes:
+        /\ log         = [i \in Servers |-> IF i \in startNodes THEN xlog({sn}, startNodes) ELSE << >>]
+        /\ state       = [i \in Servers |-> IF i = sn THEN Leader ELSE IF i \in startNodes THEN Follower ELSE None]
+        /\ commitIndex = [i \in Servers |-> IF i \in startNodes THEN Len(xlog({sn}, startNodes)) ELSE 0]
+    /\ configurations = [ i \in Servers |-> IF i \in startNodes  THEN (Len(log[i])-1 :> startNodes) ELSE << >>]
```

Comment:
This is making unspoken assumptions about what xlog constructs (end on a signature preceded by a config), which happen to be true on both StartLog and JoinedLog. I think it'd be best to at least write them in comment if we're not going to have a TypeOK for that.

Reply:
I left a comment.  Also, `SignatureInv` and `LogConfigurationConsistentInv` would be violated.

Reply:
Good point! The `SignatureInv` I had thought about, but not the `LogConfigurationConsistentInv`.